src/port1.0/portsandbox.tcl

```diff
@@ -101,19 +112,21 @@ proc portsandbox::set_profile {target} {
         lappend perms file-write-setugid
     }
 
-    # If ${prefix} is on its own volume, grant access to its
+    # If ${prefix} or the ports DB is on its own volume, grant access to its
     # temporary items directory, used by Xcode tools
-    if {[catch {get_mountpoint ${prefix_frozen}} mountpoint]} {
-        ui_debug "get_mountpoint failed: $mountpoint"
-        set mountpoint /
-    }
+    foreach dir [list ${prefix_frozen} ${portdbpath} ${portdbpath}/build] {
```

Comment:
I guess this is OK, but it also seems like someone could potentially mount pretty much any path on a different volume.

Reply:
I've expanded the check to include all immediate subdirectories of the ports DB.